python/across_api/across/resolve.py

```diff
@@ -0,0 +1,130 @@
+import json
+import logging
+from typing import Optional, Tuple
+
+import requests
+from astropy.coordinates.name_resolve import NameResolveError  # type: ignore
+from astropy.coordinates.sky_coordinate import SkyCoord  # type: ignore
+
+from ..base.common import ACROSSAPIBase
+from .schema import ResolveGetSchema, ResolveSchema
+
+ANTARES_URL = "https://api.antares.noirlab.edu/v1/loci"
+
+
+def antares_radec(ztf_id: str) -> Tuple[Optional[float], Optional[float]]:
+    """
+    Query ANTARES API to find RA/Dec of a given ZTF source
+
+    Parameters
+    ----------
+    ztf_id
+        ZTF name of source
+
+    Returns
+    -------
+        RA, Dec in ICRS decimal degrees, or None, None if not found
+
+    FIXME: Replace with antares-client module call in future, once confluent-kafka-python issues are resolved.
+    """
+    search_query = json.dumps(
+        {"query": {"bool": {"filter": {"term": {"properties.ztf_object_id": ztf_id}}}}}
+    )
+
+    params = {
+        "sort": "-properties.newest_alert_observation_time",
+        "elasticsearch_query[locus_listing]": search_query,
+    }
+    r = requests.get(ANTARES_URL, params=params)
+    try:
+        r.raise_for_status()
+    except requests.HTTPError as e:
+        logging.exception(e)
+
+    if r.ok:
+        antares_data = json.loads(r.text)
+        if antares_data["meta"]["count"] > 0:
+            ra = antares_data["data"][0]["attributes"]["ra"]
+            dec = antares_data["data"][0]["attributes"]["dec"]
+            return ra, dec
+    return None, None
```

Comment:
Ah, we can simplify this a bit. It seems that the ANTARES API will return an HTTP 200 OK status even if no match is found. So we can simply raise for any other status; raising an uncaught exception will result in Lambda returning an HTTP 500 error, which is appropriate.

```suggestion
    r = requests.get(ANTARES_URL, params=params)
    r.raise_for_status()
    antares_data = r.json()
    if antares_data["meta"]["count"] > 0:
        ra = antares_data["data"][0]["attributes"]["ra"]
        dec = antares_data["data"][0]["attributes"]["dec"]
    else:
        ra = dec = None
    return ra, dec
```